gpustack/schemas/workers.py

```diff
@@ -145,6 +145,34 @@ class WorkerBase(SQLModel):
         sa_column=Column(UTCDateTime), default=None
     )
 
+    def compute_state(self, worker_offline_timeout=60):
+        now = int(datetime.now(timezone.utc).timestamp())
+        heartbeat_timestamp = (
+            self.heartbeat_time.timestamp() if self.heartbeat_time else None
+        )
+
+        if (
+            heartbeat_timestamp is None
+            or now - heartbeat_timestamp > worker_offline_timeout
+        ):
+            self.state = WorkerStateEnum.NOT_READY
+            self.state_message = "Heartbeat lost"
+            return
+
+        if self.unreachable:
+            healthz_url = f"http://{self.ip}:{self.port}/healthz"
+            msg = (
+                f"Server cannot access the worker's healthz service at {healthz_url}. "
+                "Please verify the port requirements in the "
+                "<a href='https://docs.gpustack.ai/latest/installation/installation-requirements/'>documentation</a>"
```

Comment:
Should we add `target='_blank'`

Reply:
No need, the default behavior of the UI for this type of link is to open in a new window.